### ISSUE TITLE
Implicit flow - response type not being taken into account

### ISSUE THREAD

**Issue by whiskerch** [FusionAuth/fusionauth-issues#102]
### Description 
I am trying to get the oauth implicit flow working from an angular application using the angular-oauth2-oidc library. 

When I initiate the flow I get redirected to the URL: 
`https://identity.xxxxxxxx.io/oauth2/authorize?response_type=token%20id_token&client_id=xxxx-72dc-4535-85aa-xxxxx&state=xxxxxxxxxxxxxxxxxxxxxxxxxxx&redirect_uri=http%3A%2F%2Flocalhost%3A4200%2Fsearch&scope=openid&nonce=xxxxxxxxxxxx`

When I log in, I get returned to the url: 

`http://localhost:4200/search?code=xxxxxxxxxxxxxxxxx&state=xxxxxxxxxx&userState=Authenticated`

As far as I can see the response is the one I would expect for the response_type `code` and not `token id_token`

The application I have set up only has the `Implicit` and `Refresh Token` grants, the `Authorisuzation Code` and `Password` grants are disabled. 

### Expected behavior
The redirect upon successful login to contain an access token

**Comment by robotdan** [comment]
Hi @whiskerch thanks for letting us know. I am assuming you're on FusionAuth version 1.5.0? 

I am unable to recreate your result. Is there anything else you can share with me about your configuration that may help me identity the issue? 

If I understand your current configuration, you have configured oidc via an OAuth Angular library. When the plugin initiates login based up on your configuration you are redirected to the following FusionAuth URL: (line breaks added for readability) 

```
https://identity.xxxxxxxx.io/oauth2/authorize?
        response_type=token%20id_token
        &client_id=xxxx-72dc-4535-85aa-xxxxx
        &state=xxxxxxxxxxxxxxxxxxxxxxxxxxx
        &redirect_uri=http%3A%2F%2Flocalhost%3A4200%2Fsearch
        &scope=openid
        &nonce=xxxxxxxxxxxx
```

This all looks good, unless the `response_type` is set to `code`, we should not be returning you a `code` on the redirect. 

What configuration are you provided to the OIDC library?

**Comment by whiskerch** [comment]
I am using version 1.5.0

Even when I construct the implicit flow request url manually, I am still seeing the code response

The code that is initiating the implicit flow is: 
```
import { Component } from '@angular/core';
import { OAuthService, JwksValidationHandler } from 'angular-oauth2-oidc';
import { AuthConfig } from 'angular-oauth2-oidc';


@Component({
  selector: 'app-root',
  templateUrl: './app.component.html',
  styleUrls: ['./app.component.css']
})
export class AppComponent {
  title = 'ng-demo';

  authConfig: AuthConfig = {
    issuer: 'https://identity.xxxxxx.io',
    redirectUri: window.location.origin + '/search',
    clientId: 'xxxx-72dc-4535-85aa-xxxxx',
    strictDiscoveryDocumentValidation: false,
    jwks: {},
    oidc: true,
    scope: 'openid',
    responseType: 'token id_token',
  };

  constructor(private oauthService: OAuthService) {

    this.oauthService.configure(this.authConfig);
    // Login-Url
    this.oauthService.loginUrl = 'https://identity.xxxxxxxx.io/oauth2/authorize',

    // Use setStorage to use sessionStorage or another implementation of the TS-type Storage
    // instead of localStorage
    this.oauthService.setStorage(sessionStorage);

    this.oauthService.setupAutomaticSilentRefresh();

    // To also enable single-sign-out set the url for your auth-server's logout-endpoint here
    this.oauthService.logoutUrl = 'https://identity.xxxxxxx.io/logout';

    this.oauthService.tokenValidationHandler = new JwksValidationHandler();
    this.oauthService.tryLogin().then( res => {
      console.log('Res: ', res);
      console.log(this.oauthService.hasValidAccessToken());
      console.log(this.oauthService.getIdToken());
      console.log('state', this.oauthService.state);
      const claims = this.oauthService.getIdentityClaims();
      if (claims) {
        console.log('claims');
        console.log(claims);
      } else {
        console.log('no claims');
        //
      }
    });
    // this.oauthService.logOut();
   }

   login() {
    this.oauthService.initImplicitFlow();
   }

   logout() {
     this.oauthService.logOut();
   }
}
```
It's just a sample app - and does nothing other than test the login flow - I'm happy to zip up the code and send it over

Also - FusionAuth is in a dev environment so am happy to send any details / logs/screenshots over if you need.

**Comment by robotdan** [comment]
Yes, please do zip up the code so I can try to recreate. You can send it to daniel | @ | fusionauth | io. 

On your logout URL, I think you'll want to use the OAuth Logout to kill the SSO session.

```
this.oauthService.logoutUrl = 'https://identity.xxxxxxx.io/oauth2/logout?client_id=xxxx-72dc-4535-85aa-xxxxx';
```
Using this will log you out of the FusionAuth SSO, and then redirect to the configured Logout URL. The `client_id` is optional, but if not provided the Logout URL configured in the system wide OAuth configuration will be used. If you only have one logout URL, this will be just fine. 

https://fusionauth.io/docs/v1/tech/oauth/endpoints#logout

**Comment by robotdan** [comment]
Thanks @whiskerch for sending me your project offline. I have identified the issue. 

To work around the issue, disable self-service registration. The fix will be included in the upcoming release. 

Thanks for reporting the bug!

**Comment by robotdan** [comment]
Fixed in 1.6.0
